### ISSUE TITLE
Installation on shared webhosting

### ISSUE THREAD

**Issue by kaktux** [wallabag/wallabag#3735]
So - I got a "normal" webhoster (Meaning: can create databases, subdomains, phpmyadmin, choose php-version - all via hosting control panel (in my case plesk onyx)), Basically i can upload the files via ftp where i want to install wallabag on my subdomain (wallabag.mydomain.com) .
So no command line or tinkering/editing php, apache and stuff like that.

All other self hostable programms i know can easily be installed by
a) download program
b) copy/move/unpack onto webfolder via ftp
c) browse to webfolder where you just moved the files into -> an installer or similar occurs
done

I read and tried probably all howtos you can find on the web now - but i still don't know how to get this done on normal webhosting.
According to my hoster my webspace fullfills all requirements for wallabag - and as this was true for other programs i have there this should be right (next cloud, fresh rss).

I found the [same question asked 3 years ago](https://github.com/wallabag/wallabag/issues/1479) - but nowhere in that thread the actual question of the first post was answered. Docker, Softacolous etc. is not per se available on any shared hoster - somehow the first question drifted afar with the first answer.

Can anyone give hints (or a step by step for dummies ;) ) how (if possible) to install wallabag on a simple webhoster?

**Comment by kaktux** [comment]
like i said: i tried all howtos i could find - with the official one of course beeing the first in line.

At least i can't really figure out what is meant after downloading and extracting.
I don't know how to find/access the mentioned register form, nor how to run some query in "my favourite dbms".

**Comment by 4oo4** [comment]
So, the register form you're looking for I think is referring to this to create the first user:

![selection_003](https://user-images.githubusercontent.com/15698835/46575600-bd73bb80-c9a7-11e8-8304-5625dd1a9655.png)

If you don't have that register button, there's an option for that in `app/config/parameters.yml` that you need to set to true:
`    fosuser_registration: true`

For running that database query, how is your database set up and what kind of access do they give you with your hosting?

**Comment by kaktux** [comment]
it is set on true:

`# This file is auto-generated during the composer install
parameters:
    database_driver: pdo_mysql
    database_driver_class: null
    database_host: 127.0.0.1
    database_port: null
    database_name: wallabag
    database_user: root
    database_password: null
    database_path: null
    database_table_prefix: wallabag_
    database_socket: null
    database_charset: utf8mb4
    domain_name: 'https://your-wallabag-url-instance.com'
    mailer_transport: smtp
    mailer_host: 127.0.0.1
    mailer_user: null
    mailer_password: null
    locale: en
    secret: ovmpmAWXRCabNlMgzlzFXDYmCFfzGv
    twofactor_auth: true
    twofactor_sender: no-reply@wallabag.org
    fosuser_registration: true
    fosuser_confirmation: true
    from_email: no-reply@wallabag.org
    rss_limit: 50
    rabbitmq_host: localhost
    rabbitmq_port: 5672
    rabbitmq_user: guest
    rabbitmq_password: guest
    rabbitmq_prefetch_count: 10
    redis_scheme: tcp
    redis_host: localhost
    redis_port: 6379
    redis_path: null
    redis_password: null
`
So to sum it up - until now (after checking with my hoster that all dependencies are fullfilled) i only:
a) created a subdomain via plesk onyx control panel (wallabag.mydomain.com)
b) created a mysql database via plesk onyx control panel. I can access that via phpmyadmin (which was not needed for creation)
c) downloaded https://wllbg.org/latest-v2-package
d) extracted it to wallabag.mydomain.com

thats all.
As i said - it sounds to me like after that you simply have to browse to some page/file on wallabag.mydomain.com - and you get a gui to go on. Like the one you linked with that picture.
But thats not the case for me.

**Comment by Clindbergh** [comment]
I have the same issue. I've just tried to set up wallabag and it appears, that the documentation on [Gitbook](https://doc.wallabag.org/en/admin/installation/installation.html) are very much lacking. Or I've missed something. 

I am on a shared hosting with limited console access including composer. Steps I've done:

1. `wget https://wllbg.org/latest-v2-package && tar xvf latest-v2-package` 
2. `composer install`
3. Created a database and edited my `app/config.parameters.yml` database user, password, port, name and host
4. Edited `app/config.parameters.yml` for domain name.
5. Run ` php bin/console wallabag:install --env=prod` and `php bin/console cache:warmup --env=prod` because browsing to my wallabag installation tells me, that no database schema is there. 

# Now browsing to my installation (web dir) returns this:
**`Fatal error: Uncaught Symfony\Component\Validator\Exception\MappingException: The mapping file "/wallabag.mydomain.com/vendor/symfony/symfony/src/Symfony/Component/Form/Resources/config/validation.xml" does not exist in /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/vendor/symfony/symfony/src/Symfony/Component/Validator/Mapping/Loader/FileLoader.php:38 Stack trace: #0 /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/vendor/symfony/symfony/src/Symfony/Component/Validator/ValidatorBuilder.php(277): Symfony\Component\Validator\Mapping\Loader\FileLoader->__construct('/wallabag.mydom...') #1 /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/vendor/symfony/symfony/src/Symfony/Component/Validator/ValidatorBuilder.php(303): Symfony\Component\Validator\ValidatorBuilder->getLoaders() #2 /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/var/cache/prod/appProdProjectContainer.php(5926): Symfony\Component\Validator\ValidatorBuilder->getValidator() #3 /home/httpd/vhosts/mydomain.com/wal in /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 101  `**

How can it be resolved? What additional steps have to be taken?

- - -
Perhaps the documentation could be improved:

- Information for shared hosting: All the steps done with the php console and database setup would have to be executed locally, then uploaded to the serve respectively and the mysql dump exported locally and imported on the server.
- It is not clear how to generate the schema
- It is not clear what has to be edited in `parameters.yml`
- The order of of the required steps is entirely unclear (e.g. the doc should say: 1. `wget`, 2. Edit `parameters.yml` (...) 3. `composer install` 4. ? )

**Comment by j0k3r** [comment]
I agree the document might be improved. Feel free to open a PR with some suggestions.

Regarding your problem it's hard to diagnose without configuration information.
- where did you defined `wallabag.mydomain.com`?
- what's your `app/config/parameters.yml`?
- what's the vhost configuration?

**Comment by miwasp** [comment]
As i also like to know, how to install wallabag on shared hosting, i would like to come back to the original question:

a) created a subdomain via plesk onyx control panel (wallabag.mydomain.com)
b) created a mysql database via plesk onyx control panel. I can access that via phpmyadmin (which was not needed for creation)
c) downloaded https://wllbg.org/latest-v2-package
d) extracted it to wallabag.mydomain.com

If at this point i set
- database_host
- database_name
- database_user
- database_password
in parameters.yml i still get an emtpy page because the database is still empty. How can i create the tables, indexes, ... in the database (remember: no cmd accesss on shared hosting)?

**Comment by yomli** [comment]
I successfully installed Wallabag both on local XAMP and shared webhosting (Grandi Simple Hosting) *in a subfolder and with SQLite*. It took me three hours to understand how to do it. I wrote a simple tutorial but I'm not currently at home, so this comment is a placeholder to bring you hope: *it's not easy but it may be done*. I will post the detailed tutorial when I get home.

If I recall correctly, it involves installing [2.1.5](https://static.wallabag.org/releases/) which is one of the last version you can install with SQLite, create the account on it. Then migrate to 2.2.0 as explained [in the documentation](https://web.archive.org/web/20181119233025/https://doc.wallabag.org/en/admin/upgrade.html) (which is unavailable now with an error 500, fortunately I wrote the SQLite migration script in my tutorial so it's not lost). Then upgrade to 2.3.x.

Eventually, Wallabag will [drop support of SQLite](https://github.com/wallabag/wallabag/issues/2766) so my setup isn't future-proof.

<rant>
It's a shame Wallabag goes the same way as many Framasoft tools. It's a trend this days: you **must** have a dedicated server or cloud or whatever that can host big and bloated apps coded in many languages you will not be able to support. They claim they don't want no GAFAM, a decentralized web, but all they do is creating more and more hubs we depend on. Something as simple as an url shortener is 500ko Python with SQLite, I recreated it entirely in PHP in less than 50ko, working on every free shared webhosting. There is something mad about this trend. Even the PHP apps tend to be bloated with PHP frameworks and require 30min to an hour to upload on a SFTP because of *thousands* of files.
</rant>

**Comment by virtadpt** [comment]
@yomli This.  So much this.  We shouldn't have to maintain our own build servers just to deploy PHP applications.  I had to stand one up solely to use Wallabag and Monica.

**Comment by techexo** [comment]
Whowhowho guys (or girls)! I think we can all agree that the doc isn't perfect, and as @j0k3r said, you are welcome to do pull requests to wallabag/doc to improve it. (it's just Markdown, pretty easy to work with.)

I personally never had any issue deploying wallabag on dedicated or self-hosted servers, so this is all very configuration dependant issues. I don't have access to shared hosting so I won't be able to help. However, some common issues occurring after the installation are due to the cache: moving the installation folder in a local installation breaks everything, so I reckon it will be the same if you move to another server altogether. You might want to try emptying the cache folder after moving.

It is to note however that it's explicitly written in the documentation that deploying wallabag in a subfolder is not officially supported.

Finally, I agree with @yomli that "bloated" apps are not perfect, but they are here... and that's already that. And if a whole lot of dependencies is used, it's mostly because someone didn't have time to extract just what was needed. But feel free to fork and revamp everything, that's why GitHub exists. 😉

**Comment by techexo** [comment]
Note: I shouldn't answer at night, I'm grumpy...
For the doc part, don't hesitate to participate, even if it's only in French or English, some of us can translate it afterwards if necessary. You could also create an issue in wallabag/doc with what manipulation worked for you: having an outline for someone else to write is already a step in the good direction 😃

**Comment by techexo** [comment]
@kaktux @yomli @virtadpt @miwasp 
I might have time to work on the documentation in a few days. However, as I always self-host things, I am not really aware of what the restrictions are for the other cases.

Could you make a list of the restrictions you encounter with your host(s)? (access to SSH, php modules needed, possibility to have composer, access to database - phpmyadmin or just mysql import - and so on.)

Thanks

**Comment by virtadpt** [comment]
Modules needed can definitely be a stumbling block.  My provider (Dreamhost) makes it possible to enable them by SSHing in and editing a hidden config file.

A bigger problem is not having access to Composer.  While it is possible to download Composer and run it locally, CPU runtime limits are set for every user's account so Composer tends to get killed by the kernel after 60 seconds.  Re-running Composer over and over until it's done tends to result in broken applications.

My workaround is to maintain a Git checkout on my laptop, build Wallabag at home, then tar up the whole repo and upload it to Dreamhost.

**Comment by j0k3r** [comment]
But why do you need composer?
The archives we provide (https://static.wallabag.org/releases/) already have deps packaged. So you don't have to `composer install` yourself.

**Comment by techexo** [comment]
@j0k3r I think your answer is actually symptomatic of the issue described for the doc 😏. If it's not clear that you don't need composer when you're trying to put wallabag on shared hosting, maybe there is a problem.

I know in the doc is written
>With this package, wallabag doesn't check for mandatory extensions used in the application (theses checks are made during composer install when you have a dedicated web server, see above).

but on the requirements page
> These libraries must be installed with a tool called Composer. You need to install it if you have not already done so and be sure to use the 1.2 version (if you already have Composer, run a composer selfupdate).

So I can understand the frustration, misunderstandings and issues occurring during some installations. I'll try to revamp the Installation parts in the coming weeks (with an "s"). I'm on my way to create an issue on wallabag/doc for this, as we're polluting this feed. Everyone, feel free to go there and participate with all your issues, workarounds needed, limitations etc.

**Comment by virtadpt** [comment]
I didn't know there were pre-built packages.  The installation document issue is pretty much it - I thought I had to check it out and build it myself, so I did.  I thought it was kind of a weird flex that it was "DIY or nothing," but there are other applications out there like that so it didn't strike me as all that odd.

**Comment by edern-appere** [comment]
Hi there,

I've had the same issues a year ago when I discovered Wallabag and tried to install it on 1and1 shared webhosting. I finally gave up trying to install it and started using Pocket instead. It's a shame because Wallabag (and many other open source tools) are meant to help people become independant from web service providers but requires skills above the average in order to make them work.

I'm not here to critize so I'll get to the point : on many shared hosting, you can only drop files (SFTP), make basic database operations (via PhpMyAdmin in most cases) and choose Php version. 
What you can't do is : get command line access and edit virtual hosts. You cannot edit any Apache configuration except with some .htaccess file which is often limitated by hosting company.

So, for example if I want to install latest version of Wallabag, I can get the packages on my local computer, edit parameters.yml, then upload files to my shared hosting. I cannot go on to next step which is configuring Virtual host...

I use many web apps on my shared hosting (Nextcloud for example) and they do not require setting up a Virtual host, so it's definitely easier to install.

As for helping with the documentation, I understand that Wallabag developpers cannot do everything : write the code, test on many different environments, and write the documentation... But devs are the ones that master how Wallabag works so it's also easier for them to write the documentation. I

**Comment by techexo** [comment]
@edern-appere I began to work on a revamp of the install documentation (https://github.com/techexo/doc/tree/install/en/admin/installation), but as I do not have much time right now it's not going very fast. But this issue is definitely not abandoned!

Thank you for detailing what you can or can't do, I'll try to integrate that on the "web hosting" version of  the installation doc. I set up everything on my computer to tests different options with different servers, with or without console access and so on. 

_(Note: when asked about limitations of web hosting platforms, I had only one answer before yours... We can write documentation, but we need people to tell us what is absolutely needed vs. what's just convenient (and by we, I mean doc writers, I'm definitely not a wallabag's dev!))._

Finally, as I self-host quite many services on my server, I am a bit astonished by your comment about virtualhosts... It's often more complicated with nginx that Apache2, but lots of applications need some kind of tweaking with sites configuration (including Nextcloud if you want to use http/2, gzip compression and php's cache, for example).

The one annoying thing that I hate (strong word, but limited vocabulary...) with wallabag is the obligation to be at the root of your (sub)domain, and I'm also looking into how to allow for installation in a subdir.

Anyway, and as always: contributions welcomed and patience required 😄

**Comment by ADIX7** [comment]
Will subfolder installation be supported?

**Comment by Guduche** [comment]
Hi, I have the same problem that has Clindbergh, does someone knows what to do ?

I also tried `bin/console assets:install` for symfony but it says that it's ok.

Clindbergh's issue :
> I have the same issue. I've just tried to set up wallabag and it appears, that the documentation on [Gitbook](https://doc.wallabag.org/en/admin/installation/installation.html) are very much lacking. Or I've missed something.
> 
> I am on a shared hosting with limited console access including composer. Steps I've done:
> 
> 1. `wget https://wllbg.org/latest-v2-package && tar xvf latest-v2-package`
> 2. `composer install`
> 3. Created a database and edited my `app/config.parameters.yml` database user, password, port, name and host
> 4. Edited `app/config.parameters.yml` for domain name.
> 5. Run ` php bin/console wallabag:install --env=prod` and `php bin/console cache:warmup --env=prod` because browsing to my wallabag installation tells me, that no database schema is there.
> 
> # Now browsing to my installation (web dir) returns this:
> **`Fatal error: Uncaught Symfony\Component\Validator\Exception\MappingException: The mapping file "/wallabag.mydomain.com/vendor/symfony/symfony/src/Symfony/Component/Form/Resources/config/validation.xml" does not exist in /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/vendor/symfony/symfony/src/Symfony/Component/Validator/Mapping/Loader/FileLoader.php:38 Stack trace: #0 /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/vendor/symfony/symfony/src/Symfony/Component/Validator/ValidatorBuilder.php(277): Symfony\Component\Validator\Mapping\Loader\FileLoader->__construct('/wallabag.mydom...') #1 /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/vendor/symfony/symfony/src/Symfony/Component/Validator/ValidatorBuilder.php(303): Symfony\Component\Validator\ValidatorBuilder->getLoaders() #2 /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/var/cache/prod/appProdProjectContainer.php(5926): Symfony\Component\Validator\ValidatorBuilder->getValidator() #3 /home/httpd/vhosts/mydomain.com/wal in /home/httpd/vhosts/mydomain.com/wallabag.mydomain.com/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 101 `**
> 
> How can it be resolved? What additional steps have to be taken?
> 
> Perhaps the documentation could be improved:
> 
> * Information for shared hosting: All the steps done with the php console and database setup would have to be executed locally, then uploaded to the serve respectively and the mysql dump exported locally and imported on the server.
> * It is not clear how to generate the schema
> * It is not clear what has to be edited in `parameters.yml`
> * The order of of the required steps is entirely unclear (e.g. the doc should say: 1. `wget`, 2. Edit `parameters.yml` (...) 3. `composer install` 4. ? )

**Comment by andrewchenshx** [comment]
does installation on shared host work? 
following the latest doc:
a) downloaded https://wllbg.org/latest-v2-package
b) extracted it to a folder
c) setup a virtual host
while now , i'm curious how/when is the database schema created?

**Comment by cukabeka** [comment]
Same here. As I read from the great post by @yomli, it's totally possible to install it on shared hosting without docker. I would love to see an install script that makes it possible to install this amazing script on john doe's average webhosting. Just something oldschool, non-fancy that creates a database after putting the mySQL credentials somewhere. I googled around, but found nothing about that, only the fancy docker/composer stuff. 
Did anybody manage to create something like this and install it on shared hosting?

**Comment by cukabeka** [comment]
I actually got it running by 
- editing the variables in parameters.yml 
- and choosing a short password for the mysql database (complex PWs did not work) + allowing external access to the database ("localhost" was not possible for accessing the database, I had to set **database_host** to https://mydomain.com/)
Now I get: 
`Base table or view not found: 1146 Table 'db_name.wallabag_craue_config_setting' doesn't exist`

At this point a SQL dump of a fresh install would be extremely helpful so that I could just import it in phpMyAdmin, and I guess I would be ready to go.

Can anybody provide such a SQL dump? Is it hidden somewhere in the pre-built package?

**Comment by mbirth** [comment]
Have you guys tried this:

    bin/console --env=prod list

**Comment by andrewchenshx** [comment]
after several days' strugling,, it seems work!
a) downloaded https://wllbg.org/latest-v2-package
b) extracted it to a folder
c) config the /app/config/parameters.yml
caution: the marialdb password should be **single quoted**.
eg:     database_password: '#MyPassword1357'
d) now the magic part(which i think do the creation of the database)
php bin/console cache:clear --env=prod
php bin/console --env=prod
php bin/console wallabag:install --env=prod

the only problem, which i find in the log file:
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP Warning:  "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /volume1/web/wallabag2.3.8/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php on line 2665
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP Stack trace:
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   1. {main}() /volume1/web/wallabag2.3.8/web/app.php:0
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   2. AppKernel->handle() /volume1/web/wallabag2.3.8/web/app.php:18
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   3. Symfony\Component\HttpKernel\HttpKernel->handle() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:169
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   4. Symfony\Component\HttpKernel\HttpKernel->handleRaw() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   5. Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher->dispatch() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:127
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   6. Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher->getListeners() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:43
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   7. Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher->getListeners() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/ContainerAwareEventDispatcher.php:128
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   8. Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher->sortListeners() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:61
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP   9. appProdProjectContainer->{closure:/volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:1567-1569}() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:229
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  10. appProdProjectContainer->get() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:1568
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  11. appProdProjectContainer->getNelmioApiDoc_EventListener_RequestService() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:329
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  12. appProdProjectContainer->get() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:3853
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  13. appProdProjectContainer->getNelmioApiDoc_Formatter_HtmlFormatterService() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:329
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  14. appProdProjectContainer->get() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:3896
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  15. appProdProjectContainer->getTemplatingService() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:329
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  16. appProdProjectContainer->get() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:5344
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  17. appProdProjectContainer->getTwigService() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:329
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  18. appProdProjectContainer->get() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:5776
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  19. appProdProjectContainer->getWallabagCore_EntryRepositoryService() /volume1/web/wallabag2.3.8/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:329
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  20. DoctrineORMEntityManager_0000000036b1cd2e00000000105c78c29820a9b9ecc2bffda159b11bb497035f->getRepository() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:6124
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  21. Closure->__invoke() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:9117
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  22. appProdProjectContainer->{closure:/volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:1231-1237}() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:9117
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  23. appProdProjectContainer->getDoctrine_Orm_DefaultEntityManagerService() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:1232
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  24. Doctrine\ORM\EntityManager::create() /volume1/web/wallabag2.3.8/var/cache/prod/appProdProjectContainer.php:1274
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  25. Doctrine\ORM\EntityManager->__construct() /volume1/web/wallabag2.3.8/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:850
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  26. spl_autoload_call() /volume1/web/wallabag2.3.8/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:160
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  27. Composer\Autoload\ClassLoader->loadClass() /volume1/web/wallabag2.3.8/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:160
[04-Apr-2020 12:42:00 Asia/Shanghai] PHP  28. Composer\Autoload\includeFile() /volume1/web/wallabag2.3.8/vendor/composer/ClassLoader.php:322

**Comment by cukabeka** [comment]
Hi @andrewchenshx - amazing, thanks! That really helped!

So here's my personal recipe, remixed with @andrewchenshx 's wisdom (please steal it for the docs):

### If you have SSH access: 

> 1. download https://wllbg.org/latest-v2-package
> 2. extracted it to a folder on your server
> 3. config the `/app/config/parameters.yml` . caution: the marialdb password should be single quoted, eg: database_password: '#MyPassword1357'. Make sure you just add **localhost** (no parantheses) as database_host.
> 4. now the magic SSH part (for the creation of the database). Enter this in your terminal
> ```
> php bin/console cache:clear --env=prod
> php bin/console --env=prod
> php bin/console wallabag:install --env=prod
> ```

Please note that if anything goes wrong with the credentials in parameters.yml, you need to clear the cache again with **php bin/console cache:clear --env=prod**


### If you don't have SSH: 

> 1. Follow the first 3 steps above.
> 4. Import the blank database scheme dump in phpMyAdmin from my GIST over here: https://gist.github.com/cukabeka/8ca3789bca0f8d3770c6e3481ddd564d 
> user/password stored in the SQL is wallabag/wallabag
> Now wallabag should be installed. You can now configure your web server, see https://doc.wallabag.org

**Comment by qwertygc** [comment]
@cukabeka  Think a lot ! I have searched desperately how to install in a shared host and your solution work. Maybe write in the documentation your content and mysql snippet.

**Comment by ArmandPhilippot** [comment]
Thanks @cukabeka ! I was able to install Wallabag on Gandi Simple Hosting.
However, is the second command really useful? (I'm not familiar with Symfony) With or without, it works with my shared hosting.

Like @andrewchenshx, I have a "PHP Warning" when I enter the 2nd command (or the 3rd if I omitted the 2nd).
```
PHP Warning:  "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /srv/data/web/vhosts/subdomain.domainname.com/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php on line 2636
```

Despite that, everything works!

**To add my small contribution:**
If your host has a public folder (`htdocs`, `www` ...), you must create a symbolic link between the `web` folder and this folder.
(I placed all Wallabag files at the root of my subdomain, I deleted the existing `htdocs` folder and created the symbolic link.)

@edern-appere Yes, it works. I did not need to configure a virtual host. But, as I indicate above, you may need a symbolic link depending on your host.

**Comment by msrn** [comment]
With @cukabeka recipe I'm getting error upon database creation

` SQLSTATE[42S02]: Base table or view not found: 1146 Table 'dbname.wallabag_
  user' doesn't exist
`
All credentials are okay. I have MariaDB but setting localhost as database_host was not suitable.

**Comment by GiovanniSalmeri** [comment]
I have published a little recipe at https://gist.github.com/yomli/cc4185eda47a929ab2e314ac8cce6425#gistcomment-3586856
Perhaps somebody will find it useful.

**Comment by rodrigoawk** [comment]
> 
> 
> I successfully installed the 1) latest release of Wallabag (2.4.1) 2) in a shared webhosting 3) without SSH access 4) in un subfolder 4) with SQLite. Since I think that this is a rather common situation, this is my little and easy recipe. The same recipe should work for MySQL _without using PHPMyadmin_, with the right parameters in step 2. Obviously, YMMV and corrections are much welcomed.
> 
>     1. Upload `wallabag-release-2.4.1.tar.gz` and decompress it in a folder (let's say `/wallabag/`) with the file manager of the control panel of your webhosting.
> 
>     2. Modify the file `/wallabag/app/config/parameters.yml` following the page https://doc.wallabag.org/en/admin/parameters.html. In particular the first lines must be as follows (replace `mydomain.org` with your domain name):
> 
> 
> ```
> 	database_driver: pdo_sqlite
> 	database_host: 127.0.0.1
> 	database_port: null
> 	database_name: wallabag
> 	database_user: root
> 	database_password: null
> 	database_path: %kernel.root_dir%/../data/db/wallabag.sqlite
> 	database_table_prefix: wallabag_
> 	database_socket: null
> 	database_charset: utf8mb4
> 	domain_name: 'https://mydomain.org/wallabag'
> ```
> 
>     1. Upload to `/wallabag/` a file named `install.php` with the following content:
> 
> 
> ```
> <?php
> exec('bin/console wallabag:install --env=prod --no-interaction', $output, $retval);
> header("Content-Type: text/plain"); 
> if ($retval !== 0) {
> 	echo "An error occured!\n\n";
> 	echo "Error code: $retval\n\n";
> 	echo "Error message:\n\n";
> }
> echo implode("\n", $output);
> ```
> 
>     1. Visit with your browser the address https://mydomain.org/wallabag/install.php. If all goes right, you should read the following messages:
> 
> 
> ```
> wallabag installer
> ==================
> 
> Step 1 of 4: Checking system requirements.
> ------------------------------------------
> 
>  ------------------------- -------- ----------------
>   Checked                   Status   Recommendation
>  ------------------------- -------- ----------------
>   PDO Driver (pdo_sqlite)   OK!
>   Database connection       OK!
>   Database version          OK!
>   curl_exec                 OK!
>   curl_multi_init           OK!
>  ------------------------- -------- ----------------
> 
>  [OK] Success! Your system can run wallabag properly.
> 
> Step 2 of 4: Setting up database.
> ---------------------------------
> 
>  Creating schema...
>  Clearing the cache...
> 
>  Database successfully setup.
> 
> Step 3 of 4: Administration setup.
> ----------------------------------
> 
>  Administration successfully setup.
> 
> Step 4 of 4: Config setup.
> --------------------------
> 
>  Config successfully setup.
> 
>  [OK] wallabag has been successfully installed.
> 
>  [OK] You can now configure your web server, see https://doc.wallabag.org
> ```
> 
>     1. **Delete** the file `install.php`!
> 
>     2. Upload to `/wallabag/` a file named `.htaccess` (initial dot!) with the following content:
> 
> 
> ```
> RewriteEngine On
> RewriteRule ^(.*)$ /wallabag/web/$1 [NC,L,R]
> ```
> 
>     1. Visit the address https://mydomain.org/wallabag/ and register.
> 
>     2. Use in your mobile app the address https://mydomain.org/wallabag/web
> 
>     3. Anyone who knows the address of your wallabag instance can freely register. If you want to prevent this, change in the file `/wallabag/app/config/parameters.yml` the setting `fosuser_registration` to `false`.

In order to adjust your setup instructions, I did the following:

5.2 Edit /wallabag/app/config/parameters.yml if you do not configure SMTP options - this will prevent to send/request account activation link and will activate it right away:
fosuser_confirmation: false

8.2 Edit /wallabag/web/.htaccess in order to prevent registrations - only changing "fosuser_registration: false" did not work here (maybe a bug?):
<FilesMatch ".*register$">
    Order Allow,Deny
    Deny from all
</FilesMatch>

Notices:
i- Tried both sqlite and mysql - they do work as expected.

ii- Instapaper import seems to be broken; even tried using the hosted version at wallabag.it but got the same error:
 Error while processing import. Please verify your imput file.

(doesn't seem to be a problem with uploading/folder privileges because the import function is working as expected using wallabag-v2 and .json as input file)

Not sure if it is recommended but I did chmod -R 777 to /wallabag/web/assets/ and /wallabag/web/uploads/ - still no success with Instapaper.

iii- To allow/disallow advanced config options (from the console) and logoff/logon from Wallabag to take effect:
 php bin/console fos:user:promote --env=prod --super <username>
 php bin/console fos:user:demote --env=prod --super <username>

(this is needed if you want to hide buttons like Twitter and other social media sharing from the tool bar.)

iv- Advanced config option: Internal Settings > MISC > Download images locally - default is '0'; if set to '1', after saving a new url it goes to a page error and do not save images locally:

 500: Internal Server Error
 Call to a member function bindHandles() on null

The article is saved though, but the images on it (includding the thumbnail) still point to original url.
*Another side effect from this Settings is that mobile app (Android) stopped syncing as expected.

v- A tip if an article/website is not retrieved properly is to intermediate processing thru https://www.printfriendly.com/ - it will give an extra step (manual so far) but can override fetch issues and have it saved for later!

-
Conclusion:

Migrating from Instapaper to Wallabag is a go for me, but (item iv) is really needed if you want to preserve articles for future, as this application claims to be. Also if server hosting the article prevents image-hotlink this might not work, I assume.

Help that this is useful for someone else.

>>> Does anyone have an workaround to enable saving images locally? :)

**Comment by RolandTi** [comment]
As this thread is the first response in google for installing wallabag on gandi.net, I want to complete it with the curl issue.
If you can't grab any https article and you get in the log : 
`app.ERROR: Error: cURL error 60: SSL certificate problem: certificate has expired when sending request: GET`
you can solve it by downloading [cacert.pem](https://curl.se/ca/cacert.pem) and putting it in 
`/srv/data/web/vhosts/(your wallabag instance)/cacert.pem`
edit the custom-php.ini by adding the line : 
`curl.cainfo=/srv/data/web/vhosts/(your wallabag instance)/cacert.pem`, restart the server and now you can grab any https article :-)
